### PR TITLE
feat: Default stackableVersion to operator version

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Part of https://github.com/stackabletech/issues/issues/404

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Added
+
+- Default stackableVersion to operator version ([#493]).
```
Comment (lfrancke) [inline CHANGELOG.md:9]: I'd say this needs a more prominent warning that this is a CRD change?

Comment (sbernauer) [reply CHANGELOG.md:9]: Do you have any suggestion of what it could look like? Its non-breaking btw ;)

Comment (lfrancke) [reply CHANGELOG.md:9]: I have not checked if it's valid Markdown but something along those lines.

Comment (sbernauer) [reply CHANGELOG.md:9]: WDYT of

- Default stackableVersion to operator version. It is recommended to remove `spec.image.stackableVersion` from your custom resources ([#493]).

Comment (sbernauer) [reply CHANGELOG.md:9]: As you are on vacation, and @siegfriedweber needs this PR to continue his work I would go with my proposed entry. We can continue discussing it here, after we reach a conclusion we need to change it in *all* operators anyway (so one operator more does not make a big change)

File: Cargo.lock [hunk 2]
```diff
@@ -1721,18 +1711,18 @@ dependencies = [
...
-checksum = "bebd363326d05ec3e2f532ab7660680f3b02130d780c299bca73469d521bc0ed"
+checksum = "b0293b4b29daaf487284529cc2f5675b8e57c61f70167ba415a463651fd6a918"
 dependencies = [
  "serde",
 ]
 
 [[package]]
 name = "serde"
-version = "1.0.164"
+version = "1.0.180"
```
Comment (siegfriedweber) [inline Cargo.lock:1723]: serde 1.0.172+ cannot be built with Nix (see https://github.com/serde-rs/serde/issues/2538). Please downgrade it to version 1.0.171:

```shell
cargo update --package serde --precise 1.0.171
```

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

**Review by siegfriedweber — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (sbernauer) [reply CHANGELOG.md]: I need to resolve this for merging, reopening it afterwards

**Pull request merged**